dask/utils.py

```diff
@@ -397,6 +397,43 @@ def random_state_data(n: int, random_state=None) -> list:
     return l
 
 
+def random_rng_data(n: int, random_state=None) -> list:
+    """Return a list of arrays that can initialize a ``BitGenerator`` in
+    ``np.random.Generator``.
+
+    Parameters
+    ----------
+    n : int
+        Number of arrays to return.
+    random_state : int or BitGenerator state or Generator, optional
+        If an int, is used to seed a new ``Generator``.
+    """
+    import numpy as np
+
+    if all(
+        hasattr(random_state, attr) for attr in ["normal", "beta", "bytes", "uniform"]
+    ):
+        # Generator
+        gen = random_state
+    elif isinstance(random_state, dict):
+        # BitGenerator state
+        bit_generator = np.random.PCG64()
+        bit_generator.state = random_state  # type: ignore
+        gen = np.random.Generator(bit_generator)
+    elif isinstance(random_state, (int, type(None))):
+        # int or None to be used as seed
+        gen = np.random.default_rng(seed=random_state)
+    else:
+        raise ValueError(
+            "random_state is not an int or a BitGenerator state or a Generator"
+        )
+
+    random_data = gen.bytes(624 * n * 4)  # `n * 624` 32-bit integers
```

Comment:
Why `624`? May be worth fleshing out this comment a bit more

Reply:
Well, `624` is really a left over from when Mersenne Twister was the default `BitGenerator`. `MT19937` generates random numbers in groups of `624` at a time and `MT19937` state remains the same until those `624` numbers are consumed and Twister needs to be called again. I don't see how calling Twister again and again while using `np.random.RandomState` helps us with achieving non overlapping independent random number streams but that is probably my lack of understanding.

So removed `624` since we do not use `MT19937` and instead the function now returns 64bit integers (as opposed to 32bits) to be used as seeds.

This can still be improved by using `SeedSequence.spawn()` method to generate SeedSequences but unfortunately is too invasive a change for this PR. https://numpy.org/doc/stable/reference/random/parallel.html is a good summary